### PR TITLE
feat: replacing minimap with suiteSelector component

### PR DESCRIPTION
**Pull request by nastyastavitskaya**
## Summary

- This PR introduces a new component to switch between suites - SuiteSelector in replacement to minimap.
_Pending_
- [x] Product confirmation on updating SuiteSelectir trigger from hover to click across all 3 apps.

## Testing Plan

- [x] Was this tested locally? If not, explain why.
- Run the global navigation --mp-with-nav-switcher-tour story. The story has a tour component showing on the first click and a new SuiteSelector opens on the second suite logo click. The SuiteSelector has an active and disabled (unauthorized) links.
Screenshot 2024-08-27 at 3 06 21 PM

## Reference Issue (For mParticle employees only. Ignore if you are an outside contributor)

- Closes https://mparticle-eng.atlassian.net/browse/UNI-931

### CODE REVIEW

**Comment by ilyusha** [comment]
### Review Forge
[Click Here for AI Insights](https://supreme-adventure-m612l2p.pages.github.io?prUrl=https://github.com/mParticle/aquarium/pull/383)

File: src/components/navigation/GlobalNavigation/SuiteLogo.tsx [hunk 1]
```diff
@@ -36,57 +36,43 @@ export function SuiteLogo({
...
-        placement="bottomLeft"
+      <SuiteSelector
         open={isPopoverOpen}
-        trigger="hover"
         onOpenChange={handlePopoverOpenChange}
-        arrow={false}>
-        <SuiteLogoContent />
-      </Popover>
+        onLinkClick={handleLinkClick}
+        minimapOptions={props}>
```
Comment (gabyzif) [inline src/components/navigation/GlobalNavigation/SuiteLogo.tsx:73]: should we call it `SuiteSelectorOptions`?

File: src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import {
+  type IMiniMapOptions,
+  type MiniMapLink,
+} from 'src/components/navigation/GlobalNavigation/GlobalNavigationItems'
+import { SuiteSelectorSuiteLink } from 'src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorSuiteLink'
+
+type ISuiteSelectorContentProps = IMiniMapOptions
+
+export function SuiteSelectorContent(props: ISuiteSelectorContentProps) {
+  const handleLinkClick = (link: MiniMapLink): void => {
```
Comment (gabyzif) [inline src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx:14]: nit: should we update `MiniMapLink` type for `SuiteSelectorLink` ?

**Review by gabyzif — APPROVED** [review]

File: src/components/navigation/GlobalNavigation/SuiteLogo.tsx (hunk 1 above)
Comment (nastyastavitskaya) [reply src/components/navigation/GlobalNavigation/SuiteLogo.tsx]: indeed!

File: src/components/navigation/GlobalNavigation/SuiteLogo.tsx [hunk 3]
```diff
@@ -28,69 +28,55 @@ export function SuiteLogo({
...
   navSwitcherTourOptions,
-  minimapOptions,
+  suiteSelectorOptions,
 }: SuiteLogoProps) {
   const logoRef = useRef(null)
 
-  if (!minimapOptions || navSwitcherTourOptions?.open) {
-    return <SuiteLogoContent />
+  if (!suiteSelectorOptions || navSwitcherTourOptions?.open) {
+    return <SuiteLogoContent onLogoClick={onSuiteLogoClick} />
```
Comment (gabrielmarquesf) [inline src/components/navigation/GlobalNavigation/SuiteLogo.tsx:36]: why did you add `onLogoClick={onSuiteLogoClick}` here?

File: src/components/navigation/GlobalNavigation/SuiteLogo.tsx [hunk 4]
```diff
@@ -28,69 +28,55 @@ export function SuiteLogo({
...
-          {renderNavLogo()}
+          <NavLogo onLogoClick={onLogoClick} />
           {navSwitcherTourOptions && renderNavTour(navSwitcherTourOptions)}
         </div>
       </>
     )
   }
 
-  function MinimapWithPopover(props: IMiniMapOptions) {
+  function LogoWithSuiteSelector(props: ISuiteSelectorOptions) {
```
Comment (gabrielmarquesf) [inline src/components/navigation/GlobalNavigation/SuiteLogo.tsx:52]: It's not a good practice to have component functions inside the render function, because it will create a new `LogoWithSuiteSelector` function every re-render, then react will un-mount and mount a new `LogoWithSuiteSelector` component every re-render. This can lead to some weird situations and performance issues.
Alternatives are: change it to a render function like `renderLogoWithSuiteSelector()`, or move this function to outside of render fn (usually even better to another file). I see you are using local variables inside this fn, so I would suggest the first alternative.

Comment (gabrielmarquesf) [reply src/components/navigation/GlobalNavigation/SuiteLogo.tsx:52]: ref: https://www.developerway.com/posts/react-re-renders-guide#part3.1

File: src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    const isLinkUnauthorized = props.unauthorizedLinks?.includes(link.linkId)
+
+    if (link) {
+      if (isLinkUnauthorized) props.onUnauthorizedClick?.(link)
+      else props.onLinkClick(link)
+    }
+  }
+
+  return (
+    <ConfigProvider>
```
Comment (gabrielmarquesf) [inline src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx:24]: curious, why do you need it?

File: src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+        {renderOverviewButton()}
+      </div>
+    </ConfigProvider>
+  )
+
+  function renderOverviewButton() {
+    return (
+      <>
+        <Flex align="center" justify="start">
```
Comment (gabrielmarquesf) [inline src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx:49]: I think you don't need this `Flex`, do you?

File: src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx (hunk 5 above)
Comment (nastyastavitskaya) [reply src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx:24]: good point, I think I just copied it from ex-minimap. I'll remove it

File: src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx (hunk 6 above)
Comment (nastyastavitskaya) [reply src/components/navigation/GlobalNavigation/SuiteSelector/SuiteSelectorContent.tsx:49]: flex police 🚔 😄

File: src/components/navigation/GlobalNavigation/SuiteLogo.tsx (hunk 3 above)
Comment (nastyastavitskaya) [reply src/components/navigation/GlobalNavigation/SuiteLogo.tsx:36]: so the idea is: if there is no suite selector, I want to pass the onSuiteLogoClick to the component (like in analytics it's used to navigate to home page). but if there is a suite selector, the click should not be the prop but the functionality to show/hide the suite selector

**Review by gabrielmarquesf — APPROVED** [review]

**Review by gabyzif — APPROVED** [review]
looking good!
